service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/TxBatchHandler.kt

```diff
@@ -0,0 +1,99 @@
+package io.provenance.digitalcurrency.consortium.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import cosmwasm.wasm.v1.Tx
+import io.provenance.digitalcurrency.consortium.config.BankClientProperties
+import io.provenance.digitalcurrency.consortium.config.ProvenanceProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.domain.AddressDeregistrationRecord
+import io.provenance.digitalcurrency.consortium.domain.AddressRegistrationRecord
+import io.provenance.digitalcurrency.consortium.domain.CoinBurnRecord
+import io.provenance.digitalcurrency.consortium.domain.CoinMintRecord
+import io.provenance.digitalcurrency.consortium.domain.CoinRedemptionRecord
+import io.provenance.digitalcurrency.consortium.domain.TxRequestType
+import io.provenance.digitalcurrency.consortium.domain.TxRequestViewRecord
+import io.provenance.digitalcurrency.consortium.domain.TxStatus
+import io.provenance.digitalcurrency.consortium.extension.getAddAttributeMessage
+import io.provenance.digitalcurrency.consortium.extension.getDeleteAttributeMessage
+import io.provenance.digitalcurrency.consortium.extension.getExecuteContractMessage
+import io.provenance.digitalcurrency.consortium.extension.toByteString
+import io.provenance.digitalcurrency.consortium.messages.ContractMessageI
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+
+@Service
+class TxBatchHandler(
+    private val pbcService: PbcService,
+    private val provenanceProperties: ProvenanceProperties,
+    private val bankClientProperties: BankClientProperties,
+    private val mapper: ObjectMapper,
+    private val pbcTimeoutService: PbcTimeoutService,
+) {
+    private val log = logger()
+
+    @Scheduled(
+        initialDelayString = "\${queue.batch_initial_delay_ms}",
+        fixedRateString = "\${queue.batch_polling_delay_ms}"
+    )
+    fun batchQueuedTxns() {
+        transaction {
+            TxRequestViewRecord.findQueued().mapNotNull { request ->
+                when (request.type) {
+                    TxRequestType.MINT -> CoinMintRecord.findById(request.id)!!.let {
```

Comment:
wondering if each of these records needs to be forUpdate locked?

Reply:
well maybe not; i'm slightly wary that we have everything in a single transaction? wondering if its better to separate the message generate, and the status updates into separate tx's